presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -856,6 +858,30 @@ public void commitInsert(ConnectorSession session, ConnectorInsertTableHandle in
             for (CompletableFuture<?> fileRenameFuture : fileRenameFutures) {
                 MoreFutures.getFutureValue(fileRenameFuture, PrestoException.class);
             }
+
+            if (handle.getWritePath().isPresent()) {
+                // clean remaining partition directories
+                String insertWritePath = handle.getWritePath().get();
+                partitionUpdates.stream()
+                        .map(PartitionUpdate::getWritePath)
+                        .distinct()
+                        .forEach(writePath -> {
+                            checkState(writePath.startsWith(insertWritePath), "Partition updated was write stored outside insert write path");
+                            while (!writePath.equals(insertWritePath)) {
+                                if (!deleteIfExists(writePath)) {
+                                    // this is temp data so an error isn't a big problem
+                                    log.debug("Error deleting insert temp data in %s", writePath);
```

Comment:
This is not an error. You are deleting parents in a loop. At some point, you are going to fail, and it is expected.

For example, there are 2 write paths: `/a/b/c` and `/a/b/d`. After you delete `/a/b/c`, you'll try deleting `/a/b`. And this will fail because `/a/b/d` hasn't been deleted.